webapp/channels/src/components/post_view/gm_conversion_message/gm_conversion_message.tsx

```diff
@@ -36,8 +36,9 @@ function GMConversionMessage(props: Props): JSX.Element {
         ),
     );
 
-    const convertedByUserUsername = userProfiles.find((user) => user.id === convertedByUserId)!.username;
-    const gmMembersUsernames = userProfiles.map((user) => renderUsername(user.username));
+    // we can memoize on profile length as the number of users in a GM always stay constant
+    const convertedByUserUsername = useMemo(() => userProfiles.find((user) => user.id === convertedByUserId)!.username, [userProfiles.length]);
```

Comment:
Memoizing on the length you are losing information about the users. If the user changes its username, it would not be reflected here.

Reply:
Removed memomization as discussed.